pandas/io/formats/csvs.py

```diff
@@ -115,7 +122,7 @@ def __init__(self, obj, path_or_buf=None, sep=",", na_rep='',
         self.data_index = obj.index
         if (isinstance(self.data_index, (ABCDatetimeIndex, ABCPeriodIndex)) and
                 date_format is not None):
-            from pandas import Index
+            from pandas import Index  # type: ignore
```

Comment:
What error was this giving you?

Reply:
Giving `pandas/io/formats/csvs.py:125: error: Module 'pandas' has no attribute 'Index'`; It appears to only occur if there are type annotations on any `CSVFormatter` parameters.

Reply:
Make sure you try again on master - I think a separate PR should have resolved this already

Reply:
The same error appears on master if any type annotations are added.

Reply:
@drew-heenan can you remove this and push as a new commit? Again thought we resolved this in a separate PR so would like to validate its not a mypy versioning thing between your local environment and what we have on CI

Reply:
@WillAyd Just did that - the error still appears.

Reply:
Hmm OK thanks for confirming. @ryankarlos not sure if you have any insight - thought this would be resolved by #26019 

@drew-heenan this isn't a blocker so OK to add back in the ignore I think; can review separate from this

Reply:
@WillAyd Got it, thanks for checking!